Application/Frontend/components/FriendsColumn/AddFriendModal.tsx

```diff
@@ -1,88 +1,77 @@
+import { useState, useEffect } from 'react';
 import { useDisclosure } from '@mantine/hooks';
-import { Modal, Tooltip, ActionIcon, Text, Stack, Button, TextInput } from '@mantine/core';
-import { IconPlus } from '@tabler/icons-react';
-import { useState } from 'react';
-import { NewFriendModalProps } from '@/accordTypes';
-import { useEffect } from 'react';
+import { Modal, Text, Stack, Button, TextInput } from '@mantine/core';
+import { NewFriendModalProps } from '@/accordTypes'; // Make sure this path is correct for your project
+import { showNotification } from '@mantine/notifications';
 
-/**
- * Renders a modal that allows users to add a friend by entering their username. It handles
- * the logic for sending friend requests and updating the friend list upon a successful addition.
- * This component is responsible for managing its own state related to the modal's visibility,
- * the input value for the friend's username, and any error messages resulting from the request.
- *
- * @param {NewFriendModalProps} props - Component props.
- * @param {string} props.senderID - The unique ID of the current user sending the friend request.
- * @param {(value: number | null) => void} props.setLastFetched - Function to update the last fetched timestamp
- *        in the parent component, triggering a re-fetch of the friend list.
- */
 export function AddFriendModal({ senderID, setLastFetched }: NewFriendModalProps) {
   const [opened, { open, close }] = useDisclosure(false);
-  const [friendUsername, setFriendUsername] = useState(''); // Track the username input for adding a friend
-  const [searchResult, setSearchResult] = useState<number>();
-  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [friendUsername, setFriendUsername] = useState('');
+  const [searchResult, setSearchResult] = useState<number | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
 
-    /**
-   * Handles the logic for sending a friend request based on the entered username. It performs a POST
-   * request to the server with the sender's ID and the friend's username. Upon a successful request,
-   * it resets the input field for further requests and triggers a re-fetch of the friend list by updating
-   * the `lastFetched` state. If the request fails or if no username is entered, it sets an appropriate
-   * error message.
-   */
   const handleAddFriendClick = async () => {
-    if (friendUsername) {
+    if (friendUsername.trim()) {
```

Comment:
This will cause problems if somebody intentionally added whitespace in their username and this type of check is not consistent throughout all other components of the application that rely on username checks. Instead of using `.trim` everywhere, we should ensure users cannot upload usernames with whitespace. However, this would be complicated as the changes in account data are done internally through the provider.